src/System.Memory/src/System/Buffers/ReadOnlyBuffer.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a buffer that can read a sequential series of <typeparam name="T">T</typeparam>.
+    /// </summary>
+    public readonly partial struct ReadOnlyBuffer<T>
```

Comment:
Why is this called "Buffer"?  Logically I don't know how to distinguish that from "Memory".  If this is about reading, should this be called a Reader or something similar?

Reply:
We had a whole meeting about this name and decided to keep the ReadOnlyBuffer\<T\>.

Reply:
Also 

> And typically "buffer" represents something contiguous, which this doesn't appear to be.

Not necessarily true. If you look at other platforms and frameworks that have this sort of type. It's usually some type of buffer. I don't think there's a hard and fast rule here.

Reply:
I think this name is terrible.

Reply:
> Not necessarily true. If you look at other platforms and frameworks that have this sort of type. It's usually some type of buffer. I don't think there's a hard and fast rule here.

In our platform, singular "buffer" means one, as highlighted by the fact that the namespace which describes stuff related to multiple buffers plural being named with the plural "buffers".  Argument names elsewhere are named "buffer" when it's a singular array or some such thing, and lists of them are called "buffers" or "bufferList" or some similar plural or collection.

Reply:
Yea, but this is a "one or many" buffer. I'm not sure there's enough examples in our platform to provide a valid comparison.  

https://netty.io/4.0/api/io/netty/buffer/ByteBuf.html

Now I wish we kept it byte specific 😝

Reply:
> Yea, but this is a "one or many" buffer

So is basically any sequence or collection or list.

> Now I wish we kept it byte specific

If it was byte-specific, I'd be perfectly happy with it being called `ReadOnlyBytes`.